fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java

```diff
@@ -1482,16 +1482,23 @@ public void cancel() {
             }
             return;
         }
-        Coordinator coordRef = coord;
-        if (coordRef != null) {
-            coordRef.cancel();
-        }
         if (mysqlLoadId != null) {
             Env.getCurrentEnv().getLoadManager().getMysqlLoadManager().cancelMySqlLoad(mysqlLoadId);
         }
         if (parsedStmt instanceof AnalyzeTblStmt || parsedStmt instanceof AnalyzeDBStmt) {
             Env.getCurrentEnv().getAnalysisManager().cancelSyncTask(context);
         }
+
+        Coordinator coordRef = coord;
+        if (coordRef != null) {
+            coordRef.cancel();
+        } else {
+            // force cancel, for scenerios when above strategies does not work, like sql error in analyze period
+            if (context != null) {
+                context.cleanup();
```

Comment:
The modification is very dangerouse. If you cleanup the context, we do not know the actual value in context and the analyzer may use it. And send some illegal plan to BE and BE may core.